src/components/BooksQuery/BooksQuery.tsx

```diff
@@ -1,13 +1,35 @@
-import { SyntheticEvent, useState } from 'react'
+/* eslint-disable prefer-const */
+import { SyntheticEvent, useEffect, useState } from 'react'
 import './BooksQuery.css'
 import { Link } from 'react-router-dom'
-import { useGetBooks } from '../../hooks/useGetBooks'
+import { useAppSelector, useAppDispatch } from '../../redux/hook'
+import { GoogleBooks, selectCachedData } from '../../redux/BookSlice'
+import { BooksApiResponse } from '../../types/GoogleBooks'
 const BooksQuery = () => {
   const [bookName, setBookName] = useState<string>('')
-  const { books, loading, error, searchBooks } = useGetBooks('react')
+  const { searchInput, FetchedBooks, loading, error } =
+    useAppSelector(selectCachedData)
+  const dispatch = useAppDispatch()
+  const [bookList, setBooks] = useState<BooksApiResponse[]>()
+  useEffect(() => {
+    const setData = async () => {
+      try {
+        if (!searchInput) {
+          const data = await dispatch(GoogleBooks('react'))
+          setBooks(data.payload as any)
+        } else {
+          setBooks(FetchedBooks)
+        }
+      } catch {
+        console.log('data')
+      }
+    }
+    setData()
+  }, [])
   const handleSubmit = async (e: SyntheticEvent) => {
     e.preventDefault()
-    bookName ? searchBooks(bookName) : alert('Please insert book name')
+    const data = await dispatch(GoogleBooks(bookName))
+    setBooks(data.payload as any)
```

Comment:
Setting the state like this is not needed

You're getting the data from the store using the selector hook